### PR TITLE
APEXCORE-458 Using the yarn configuration helper method to read the webapp address in HA scenario

### PR DESCRIPTION
**Pull request by pramodin**
@vrozov @davidyan74 please review

### CODE REVIEW

File: engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java [hunk 1]
```diff
@@ -739,13 +745,21 @@ public static InetSocketAddress getRMWebAddress(Configuration conf, String rmId)
...
     InetSocketAddress address;
     if (sslEnabled) {
       address = conf.getSocketAddr(YarnConfiguration.RM_WEBAPP_HTTPS_ADDRESS + rmId, YarnConfiguration.DEFAULT_RM_WEBAPP_HTTPS_ADDRESS, YarnConfiguration.DEFAULT_RM_WEBAPP_HTTPS_PORT);
     } else {
       address = conf.getSocketAddr(YarnConfiguration.RM_WEBAPP_ADDRESS + rmId, YarnConfiguration.DEFAULT_RM_WEBAPP_ADDRESS, YarnConfiguration.DEFAULT_RM_WEBAPP_PORT);
     }
+    if (isHA) {
+      conf.unset(ConfigUtils.RM_HA_ID);
+    }
     LOG.info("rm webapp address setting {}", address);
```
Comment (davidyan74) [inline engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java:775]: the conf.set and conf.unset combo may not have the expected behavior if getRMWebAddress() is called by multiple threads at the same time.

File: engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java [hunk 2]
```diff
@@ -739,13 +745,21 @@ public static InetSocketAddress getRMWebAddress(Configuration conf, String rmId)
 
   public static InetSocketAddress getRMWebAddress(Configuration conf, boolean sslEnabled, String rmId)
   {
-    rmId = (rmId == null) ? "" : ("." + rmId);
+    boolean isHA = (rmId != null);
+    rmId = isHA ? ("." + rmId) : "";
```
Comment (vrozov) [inline engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java]: Should modification to `rmId` be done after the call to `conf.set(ConfigUtils.RM_HA_ID, rmId)`? It is not modified in `getRMHAAddress()`.

File: engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java [hunk 3]
```diff
@@ -739,13 +745,21 @@ public static InetSocketAddress getRMWebAddress(Configuration conf, String rmId)
...
+    if (isHA) {
+      conf = getYarnConfiguration(conf);
+      conf.set(ConfigUtils.RM_HA_ID, rmId);
+    }
     InetSocketAddress address;
     if (sslEnabled) {
       address = conf.getSocketAddr(YarnConfiguration.RM_WEBAPP_HTTPS_ADDRESS + rmId, YarnConfiguration.DEFAULT_RM_WEBAPP_HTTPS_ADDRESS, YarnConfiguration.DEFAULT_RM_WEBAPP_HTTPS_PORT);
```
Comment (vrozov) [inline engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java]: rmId should not be appended to `YarnConfiguration.RM_WEBAPP_HTTPS_ADDRESS`. `YarnConfiguration.getSocketAddr()` will return value based on `RM_HA_ID`.

File: engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java (hunk 1 above)
Comment (pramodin) [inline engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java:775]: Current use is from a single thread but will make it thread safe.

File: engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java (hunk 2 above)
Comment (pramodin) [inline engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java]: Good point, yes.

File: engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java (hunk 3 above)
Comment (pramodin) [inline engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java]: You are right, thanks for catching this

**Comment by pramodin** [comment]
Incorporated comments and squashed

File: engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java [hunk 4]
```diff
@@ -731,20 +726,51 @@ public static ApplicationReport getStartedAppInstanceByName(YarnClient clientRMS
     return null;
   }
 
+  /**
+   * Return a YarnConfiguration instance from a Configuration instance
+   * @param conf The configuration instance
+   * @return The YarnConfiguration instance
+   */
+  public static YarnConfiguration getYarnConfiguration(Configuration conf)
```
Comment (vrozov) [inline engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java]: can it be private?

Comment (pramodin) [inline engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java]: Yes will change it

**Pull request merged**